### ISSUE TITLE
Convert shapefile to geojson in browser

### ISSUE THREAD

**Issue by dwarfylo** [mbloch/mapshaper#407]
Hello,

I have used the trick described here [#267](https://github.com/mbloch/mapshaper/issues/267) and the applyCommands function described [here](https://github.com/mbloch/mapshaper/wiki/Using-mapshaper-programmatically) for a while now to convert shp (and companions) files to geojson/topojson in browser but recently my code stopped working. It seems that a check has been added (runningInBrowser) to avoid reading from cache in browser. For a while, in dev, I was overriding the function but it is not possible anymore (I think?) and more importantly, very ugly. I'm not sure if it's possible and how to read from stdin when dealing with a shp (and companions) file to get around the limitation. Was it intended to prevent this workflow and, is it possible to get around it ?

Thank you

**Comment by mbloch** [comment]
Hi, thanks for reporting this.
I'm not trying to prevent this workflow, rather it sounds like I just didn't consider your use case. I'll attempt a fix, but you will probably have to help me test it.

**Comment by dwarfylo** [comment]
I will gladly help testing it, I have a running setup to follow the path I described. To give you a bit more context, I'm in an Angular2 FrontEnd, and I retrieve binary shp files from an other API and want to convert them to topojson. Here is the typescript snippet:
```const cmd = '-i file.shp -o out.topojson';
        const inputs = {'file.shp': shp};
        if (prj) {
          inputs['file.prj'] = prj;
        }
        if (dbf) {
          inputs['file.dbf'] = dbf;
        }
        if (shx) {
          inputs['file.shx'] = shx;
        }
        console.log(cmd);
        console.log(inputs);
        mapshaper.applyCommands(cmd, inputs, (err, output) => {
          if (err) {
            console.error(err);
            return;
          }
          const json = JSON.parse(output['out.topojson']);
          this.applyTopoJson(res.study, json);
        });
```

From what I've seen in mapshaper.js code, I'm entering in:
```
cli.checkFileExists = function(path, cache) {
    if (!cli.isFile(path, cache) && path != '/dev/stdin') {
      stop("File not found (" + path + ")");
    }
  };
```
But:
```
cli.isFile = function(path, cache) {
    if (runningInBrowser()) return false;
```

So I get an error "File not found". If i comment this previous line, ofc my code is back on track, but I'm sure it has other implications ...

**Comment by mbloch** [comment]
I released a new version, 0.5.6, which has a fix for your issue. It seems to me that this change should fix the problem: https://github.com/mbloch/mapshaper/blob/master/src/cli/mapshaper-cli-utils.js#L15

**Comment by dwarfylo** [comment]
Works like a charm on my end,
Thank you very much
